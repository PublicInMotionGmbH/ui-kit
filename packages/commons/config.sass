@import './helpers'
@import './utils'

// Set up configuration variables

$prefix: talixo !default

$base-size: 8px
$gutter: grid-unit(2)

// Font size

$font-size-s: grid-unit(1.5)
$font-size-m: grid-unit(1.75)
$font-size-l: grid-unit(2)
$font-size-xl: grid-unit(2.5)
$font-size-xxl: grid-unit(4)
$font-size-title: grid-unit(12)
$font-size-subtitle: grid-unit(8)
$font-size-h1: grid-unit(6)

// Spacing

$spacing-xs: grid-unit(0.5)
$spacing-s: grid-unit(1)
$spacing-m: grid-unit(2)
$spacing-l: grid-unit(3)
$spacing-xl: grid-unit(4)
$spacing-xxl: grid-unit(8)

// Font weight

$font-weight-content: 500
$font-weight-content-l: 300
$font-weight-h1: 300
$font-weight-h2: 300
$font-weight-h3: 500
$font-weight-h4: 500
$font-weight-h5: 500
$font-weight-h6: 500
$font-weight-light: 300
$font-weight-normal: 400
$font-weight-medium: 500
$font-weight-semibold: 600
$font-weight-bold: 700

// Border width

$border-width: 1px

// Border radius

$border-radius-s: grid-unit(0.25)
$border-radius-m: grid-unit(0.5)
$border-radius-pill: grid-unit(6.25)

// Shadow

$shadow-xs: 0 grid-unit(0.125) grid-unit(0.25)
$shadow-s: 0 grid-unit(0.25) grid-unit(0.5)
$shadow-m: 0 grid-unit(0.5) grid-unit(1)
$shadow-l: 0 grid-unit(1) grid-unit(2)
$shadow-xl: 0 grid-unit(2) grid-unit(4)

// Colors

$color-talixo-navy: #273440
$color-talixo-dark-navy: #000A14
$color-talixo-red: #FF4900
$color-talixo-gray: #6F7B80
$color-talixo-light-gray: #A5BFD9
$color-talixo-lighter-gray: #F0F5F7
$color-talixo-success-green: #80DE2D
$color-talixo-warning-yellow: #E1DC04
$color-talixo-error-red: #FF4900
$color-black: #4C4C4C
$color-black-hover: #666666
$color-black-active: #262626
$color-white: #FFFFFF
$color-orange-talixo: #FF4900
$color-orange-hover: #FF672C
$color-orange-active: #E54100
$color-ghost: transparent
$color-ghost-text: #7E92A6
$color-ghost-border: #A9C7D9
$color-tab-border: #DAE0E6
$color-tab-inactive: #EBEFF2
$color-pill-blue: #00DBD4
$color-pill-gray: #CFDDE6
$color-pill-green: #33B700
$color-pill-yellow: #FFA500
$color-placeholder-text: #A9C7D9
$color-disabled-text: #A5BFD9
$color-paragraphs: #4C4C4C
$color-headers-l: #FF4900
$color-headers-m: #000A14
$color-action-items: #FF4900
$color-inline-links: #000A14
$color-border: #A5BFD9

$navy: #3d4954
$navy-text: #273440
$offwhite: #adb9bc

// Set up important helpers which depends on variables

/**
 * Build selector for specific path
 *
 * i.e. .talixo-module__element from prefix('module', 'element')
 */
@function prefix($name...)
  @return unquote('.#{$prefix}-#{toString($name, '__')}')

/**
 * Build block for specific module
 */
@mixin module ($name)
  #{prefix($name)}
    @content
