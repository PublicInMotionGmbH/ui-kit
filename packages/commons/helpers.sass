/**
 * Join list with specified glue
 */
@function toString($list, $glue: '')
  $result: null

  @for $i from 1 through length($list)
    $e: nth($list, $i)
    $result: if($i != length($list), $result#{$e}#{$glue}, $result#{$e})

  @return $result
