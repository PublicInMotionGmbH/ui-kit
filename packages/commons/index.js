const config = require('./config')

/**
 * Prefix module with global name
 *
 * @param {string} module
 * @returns {string}
 */
export function prefix (module) {
  return config.prefix + '-' + module
}
