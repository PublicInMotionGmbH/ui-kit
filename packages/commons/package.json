{
  "name": "@talixo/commons",
  "version": "1.0.0",
  "description": "Talixo common configuration",
  "author": "",
  "main": "index.js",
  "license": "ISC",
  "dependencies": {
    "rollup": "^0.56.5"
  }
}
