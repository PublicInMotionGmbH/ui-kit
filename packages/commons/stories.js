import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'

/**
 * Build styles for story
 *
 * @param {object} stylesheet
 * @returns {object}
 */
const styles = stylesheet => ({
    ...stylesheet,
    infoBody: {
        ...stylesheet.infoBody,
        boxShadow: 'none',
        border: 0,
        padding: 0
    },
    propTableHead: {
        display: 'none'
    },
    source: {
        ...stylesheet.source,
        h1: {
            ...stylesheet.source.h1,
            marginBottom: '15px'
        }
    }
})

/**
 * Create factory for stories
 *
 * @param {string} name
 * @param {object} module
 *
 * @returns {function(name: string, description: string, render: function)}
 */
export function createStoriesFactory (name, module) {
    const stories = storiesOf(name, module)

    return (name, description, render) => stories.add(name, withInfo({
        styles: styles,
        header: true,
        inline: true,
        text: description
    })(render))
}

/**
 * Get description from README file
 *
 * @param {string} content
 * @returns {string}
 */
export function getReadmeDescription (content) {
    return content
        .replace(/^<p><h1[^>]*>.*<\/h1><\/p>/g, '')
        .replace(/(<p>)?<h[1-6][> ][^]*$/i, '')
}
