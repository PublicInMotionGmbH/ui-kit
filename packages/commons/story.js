import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'

import createController from './story/createController'

/**
 * Build styles for story
 *
 * @param {object} stylesheet
 * @returns {object}
 */
const styles = stylesheet => ({
  ...stylesheet,
  infoBody: {
    ...stylesheet.infoBody,
    boxShadow: 'none',
    border: 0,
    padding: 0
  },
  propTableHead: {
    display: 'none'
  },
  source: {
    ...stylesheet.source,
    h1: {
      ...stylesheet.source.h1,
      marginBottom: '15px'
    }
  }
})

/**
 * Create factory for stories
 *
 * @param {string} name
 * @param {object} module
 *
 * @returns {function(name: string, description: string, render: function)|function(name: string, description: string, render: function, infoOptions: object)|{ controlled: function(name: string, description: string, render: function, getInitialState: function, infoOptions: object)|function(name: string, description: string, render: function, getInitialState: function, infoOptions: object) }}
 */
export function createStoriesFactory (name, module) {
  const stories = storiesOf(name, module)

  function addStory (name, description, render, infoOptions = {}) {
    return stories.add(
      name,
      withInfo({
        styles: styles,
        header: true,
        inline: true,
        text: description,
        ...infoOptions
      })(render)
    )
  }

  // Create helper for controlled stories
  addStory.controlled = function addControlledStory (name, description, render, getInitialState, infoOptions = {}) {
    // Build controller renderer
    const controller = createController(render, getInitialState)

    // Exclude controller from prop types list
    return addStory(name, description, controller, {
      propTablesExclude: [ controller.Controller ],
      ...infoOptions
    })
  }

  return addStory
}

/**
 * Get description from README file
 *
 * @param {string} content
 * @returns {string}
 */
export function getReadmeDescription (content) {
  return content
    .replace(/^<p><h1[^>]*>.*<\/h1><\/p>/g, '')
    .replace(/(<p>)?<h[1-6][> ][^]*$/i, '')
}
