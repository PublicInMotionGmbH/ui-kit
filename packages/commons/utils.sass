@function grid-unit($multiplier)
  @return $multiplier * $base-size
