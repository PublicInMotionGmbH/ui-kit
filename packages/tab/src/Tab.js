import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'
import { prefix } from '@talixo/commons'

const name = prefix('tab')

/**
 * Component which represents Tab.
 *
 * @param {object} props
 * @param {string} [props.active]
 * @param {string} [props.children]
 * @param {string} [props.className]
 * @param {string} [props.id]
 * @param {string} [props.onClick]
 * @param {string} [props.style]
 * @returns {React.Element}
 */
const Tab = ({ active, children, className, id, onClick, style }) => {
  const mappedChildren = React.Children.map(children, child => {
    return typeof child === 'string' ? <span>{child}</span> : child
  })
  return (
    <div
      className={cls(name, className, {
        active: active !== false
      })}
      style={style}
      id={id}
      onClick={onClick}
    >
      {mappedChildren}
    </div>
  )
}

Tab.propTypes = {
  /** Active state */
  active: PropTypes.bool,

  /** Tab content */
  children: PropTypes.node,

  /** Additional class name */
  className: PropTypes.string,

  /** Id passed to tab */
  id: PropTypes.string,

  /** Function that runs whe tab is clicked */
  onClick: PropTypes.func,

  /** Additional styles passed to the tab */
  style: PropTypes.object
}

Tab.defaultProps = {
  active: false
}

export default Tab
