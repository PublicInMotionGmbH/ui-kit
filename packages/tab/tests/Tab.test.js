import React from 'react'
import Tab from '../src/Tab'
import { shallow, mount } from 'enzyme'
import { prefix } from '@talixo/commons'

const name = prefix('tab')

describe('<Tab />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<Tab>Tab</Tab>)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders .action correctly', () => {
    const wrapper = shallow(<Tab className='action'>Tab</Tab>)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders active correctly', () => {
    const wrapper = shallow(<Tab active>Tab</Tab>)

    expect(wrapper).toMatchSnapshot()
  })

  it('calls onClick when clicked', () => {
    const onClick = jest.fn()
    const wrapper = mount(<Tab onClick={onClick}>Tab</Tab>)

    wrapper.find(`.${name}`).simulate('click')
    expect(onClick).toHaveBeenCalledTimes(1)
    wrapper.unmount()
  })
})
