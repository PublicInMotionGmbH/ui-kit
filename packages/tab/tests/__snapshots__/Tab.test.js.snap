// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> renders .action correctly 1`] = `
<div
  className="talixo-tab action"
>
  Tab
</div>
`;

exports[`<Tab /> renders active correctly 1`] = `
<div
  className="talixo-tab active"
>
  Tab
</div>
`;

exports[`<Tab /> renders children correctly 1`] = `
<div
  className="talixo-tab"
>
  Tab
</div>
`;
